src/Driver/WpcliDriver.php

```diff
@@ -104,41 +104,50 @@ public function wpcli($command, $subcommand, $raw_arguments = [])
 
         // Build parameter list.
         foreach ($raw_arguments as $name => $value) {
-            if (is_int($name)) {
+            if (is_numeric($name)) {
                 $arguments .= "{$value} ";
             } else {
                 $arguments .= sprintf('%s=%s ', $name, escapeshellarg($value));
             }
         }
 
-        // Support WP-CLI environment alias, or path and URL.
+        // TODO: review best practice with escapeshellcmd() here, and impact on metacharacters.
+        $config = sprintf('--path=%s --url=%s', escapeshellarg($this->path), escapeshellarg($this->url));
+
+        // Support WP-CLI environment aliases.
         if ($this->alias) {
             $config = "@{$this->alias}";
-        } else {
-            // TODO: review best practice with escapeshellcmd() here, and impact on metacharactes.
-            $config = sprintf('--path=%s --url=%s', escapeshellarg($this->path), escapeshellarg($this->url));
         }
 
 
-        $wpcli_args = sprintf('--no-color --require=%1$s/WpcliLogger.php', dirname(__DIR__));
-
         // Query WP-CLI.
-        $proc = proc_open("{$this->binary} {$config} {$wpcli_args} {$command} {$subcommand} {$arguments}", [
-            1 => ['pipe','w'],
-            2 => ['pipe','w'],
-        ], $pipes);
+        $proc = proc_open(
+            "{$this->binary} {$config} --no-color {$command} {$subcommand} {$arguments}",
+            array(
+                1 => ['pipe', 'w'],
+                2 => ['pipe', 'w'],
+            ),
+            $pipes
+        );
 
-        $cmd_output = stream_get_contents($pipes[1]);
+        $stdout = trim(stream_get_contents($pipes[1]));
+        $stderr = trim(stream_get_contents($pipes[2]));
         fclose($pipes[1]);
-        $stderr = stream_get_contents($pipes[2]);
         fclose($pipes[2]);
         $exit_code = proc_close($proc);
 
-        if ($exit_code || $stderr) {
-            throw new UnexpectedValueException("WP-CLI driver query failure: {$stderr} ($exit_code)");
+        if ($exit_code || $stderr || strpos($stdout, 'Warning: ') !== false || strpos($stdout, 'Error: ') !== false) {
```

Comment:
I'll have to make sure the colourise parameter is set to disable for that to work, otherwise we'll have odd ASCII codes for terminal colours at the start of the string. I'll test later.